### PR TITLE
Add water current

### PR DESCRIPTION
**Pull request by shifatkhan**
The particle system changes its speed and size dynamically.
**Examples**:
bubble1
bubble2

Add water current particles
Change material to HDRP

### CODE REVIEW

**Review by Akisukei — APPROVED** [review]
nice nami bubbles

File: Assets/Scripts/Utils/DrawArrow.cs [hunk 1]
```diff
@@ -0,0 +1,49 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+/// <summary>
+/// Draws debug arrows for vectors.
+/// </summary>
+public static class DrawArrow
```
Comment (danielwiktorczyk) [inline Assets/Scripts/Utils/DrawArrow.cs]: don't name a class as you would a method (with a verb)

name it "GizmoArrow" , as a noun

File: Assets/Scripts/Utils/DrawArrow.cs [hunk 2]
```diff
@@ -0,0 +1,49 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+/// <summary>
+/// Draws debug arrows for vectors.
```
Comment (danielwiktorczyk) [inline Assets/Scripts/Utils/DrawArrow.cs]: don't need this summary if it's self explanatory

File: Assets/Scripts/Utils/DrawArrow.cs [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+/// </summary>
+public static class DrawArrow
+{
+    public static void ForGizmo(Vector3 pos, Vector3 direction, float arrowHeadLength = 0.25f, float arrowHeadAngle = 20.0f)
```
Comment (danielwiktorczyk) [inline Assets/Scripts/Utils/DrawArrow.cs]: use full names: position, rather than pos

File: Assets/Scripts/Utils/DrawArrow.cs [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+    {
+        Gizmos.DrawRay(pos, direction);
+
+        Vector3 right = Quaternion.LookRotation(direction) * Quaternion.Euler(0, 180 + arrowHeadAngle, 0) * new Vector3(0, 0, 1);
```
Comment (danielwiktorczyk) [inline Assets/Scripts/Utils/DrawArrow.cs]: you can use `var` for local vars

Comment (danielwiktorczyk) [reply Assets/Scripts/Utils/DrawArrow.cs]: also you can use Vector3.right instead of Vector3(0,0,1)

File: Assets/Scripts/Utils/DrawArrow.cs [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+        Vector3 left = Quaternion.LookRotation(direction) * Quaternion.Euler(0, 180 - arrowHeadAngle, 0) * new Vector3(0, 0, 1);
+        Gizmos.DrawRay(pos + direction, right * arrowHeadLength);
+        Gizmos.DrawRay(pos + direction, left * arrowHeadLength);
+    }
+
+    public static void ForGizmo(Vector3 pos, Vector3 direction, Color color, float arrowHeadLength = 0.25f, float arrowHeadAngle = 20.0f)
```
Comment (danielwiktorczyk) [inline Assets/Scripts/Utils/DrawArrow.cs]: ForGizmo is not a great method name. Should be a verb.

File: Assets/Scripts/Utils/DrawArrow.cs [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+        Gizmos.color = color;
+        Gizmos.DrawRay(pos, direction);
+
+        Vector3 right = Quaternion.LookRotation(direction) * Quaternion.Euler(0, 180 + arrowHeadAngle, 0) * new Vector3(0, 0, 1);
+        Vector3 left = Quaternion.LookRotation(direction) * Quaternion.Euler(0, 180 - arrowHeadAngle, 0) * new Vector3(0, 0, 1);
+        Gizmos.DrawRay(pos + direction, right * arrowHeadLength);
+        Gizmos.DrawRay(pos + direction, left * arrowHeadLength);
+    }
+
+    public static void ForDebug(Vector3 pos, Vector3 direction, float arrowHeadLength = 0.25f, float arrowHeadAngle = 20.0f)
```
Comment (danielwiktorczyk) [inline Assets/Scripts/Utils/DrawArrow.cs]: naming

File: Assets/Scripts/WaterCurrent.cs [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+    public Vector3 Velocity { get; private set; }
+    
+    private void Awake()
+    {
+        Velocity = transform.forward * speed;
+    }
+
+    private void OnDrawGizmosSelected()
+    {
+        DrawArrow.ForGizmo(transform.position, transform.forward * speed);
```
Comment (danielwiktorczyk) [inline Assets/Scripts/WaterCurrent.cs]: can be => one liner

File: Assets/Scripts/WaterCurrentParticles.cs [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+
+    private void Awake()
+    {
+        waterCurrent = GetComponent<WaterCurrent>();
+        particleSystem = GetComponent<ParticleSystem>();
+    }
+
+    private void Start()
+    {
+        UpdateParticleSystem();
```
Comment (danielwiktorczyk) [inline Assets/Scripts/WaterCurrentParticles.cs]: shouldn't this be called InitializeParticleSystem() if it's only ever called once, at start?

File: Assets/Scripts/WaterCurrentParticles.cs [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+    }
+
+    private void UpdateParticleSystem()
+    {
+        particleSystem.Stop();
+
+        var speed = waterCurrent.Velocity.magnitude;
```
Comment (danielwiktorczyk) [inline Assets/Scripts/WaterCurrentParticles.cs]: nice vars ;)

**Review by danielwiktorczyk — CHANGES_REQUESTED** [review]

**Review by danielwiktorczyk — APPROVED** [review]
looks good! bubbles!! 🌊

**Pull request merged**